src/blocks/custom/form/assets/form.js

```diff
@@ -115,10 +124,25 @@ export class Form {
         }
 
         this.endLoading(isComplexSuccess);
+
+        this.maybeRedirect(isComplexSuccess);
       }
     });
   }
 
+  /**
+   * Redirects user on success if needed.
+   *
+   * @param {boolean} success Is form successfully submitted
+   */
+  maybeRedirect(success) {
```

Comment:
This method name doesn't look good me. Why do you need to add `maybe`. It will redirect only if all conditions are met.

Reply:
@gabriel-glo I mean that's why it's prefixed with `maybe`. If it was called just `redirect`, it runs and it doesn't redirect the user (because some conditions aren't met) then it shouldn't really be called `redirect`. 😄

Reply:
why not call it then `onSuccessRedirect`?

Reply:
Because it handles more than success. It handles success, if the redirect should happen or if the redirect URL for success is set 😉 

I'm fine with the name.

Reply:
> why not call it then onSuccessRedirect?

It redirects only if:
- It's a success
- "Redirect on success" option is enabled for this particular form
- Redirect URL is set

So `onSuccessRedirect` would also be incorrect for what it does.

> Because it handles more than success. It handles success, if the redirect should happen or if the redirect URL for success is set 😉

Exactly.

Reply:
I mean I'm open to alternative names if that name actually describes what this function does.

Reply:
Only thing that comes to mind is `conditionalSuccessRedirect` but I don't think it's a better name than the current one.